stuff/pipelines/client_contest.py

```diff
@@ -0,0 +1,162 @@
+                        INNER JOIN
+                            Submissions s
+                        ON
+                            s.problemset_id = pp.problemset_id
+                        INNER JOIN
+                            Runs r
+                        ON
+                            s.current_run_id = r.run_id
+                        INNER JOIN
+                            Identities i
+                        ON
+                            i.identity_id = s.identity_id
+                        INNER JOIN
+                            Contests co
+                        ON
+                            pp.problemset_id = co.problemset_id
+                        WHERE
+                            co.contest_id = %s
+                            AND r.status = 'ready'
+                            AND s.type = 'normal'
+                            AND r.verdict NOT IN ('CE', 'JE', 'VE')
+                        GROUP BY
+                            s.identity_id
+                        ORDER BY
+                            contest_score DESC
+                    ) AS r
+                INNER JOIN
+                    Contests c
+                ON
+                    c.problemset_id = r.problemset_id
+                LEFT JOIN
+                    Certificates ce
+                ON
+                    r.identity_id = ce.identity_id
+                    AND c.contest_id = ce.contest_id
+                CROSS JOIN
+                    (SELECT @n := 0) AS temp
+                WHERE
+                    certificate_id IS NULL;
+                ''', (data['contest_id'],))
+        certificates = []
+        for row in cur:
+            contest_place = None
+            if row['certificate_cutoff'] is None:
+                logging.info('The contest has no places')
+            elif row['place'] > row['certificate_cutoff']:
+                logging.info('The user did not reach the place to be reported')
+            else:
+                contest_place = row['place']
+            code_verification = generate_code()
+            certificates.append((
+                row['identity_id'], 'contest', data['contest_id'],
+                code_verification, contest_place
+            ))
+        cur.executemany('''
+            INSERT INTO
+                `Certificates` (
+                    `identity_id`,
+                    `certificate_type`,
+                    `contest_id`,
+                    `verification_code`,
+                    `contest_place`)
+            VALUES(%s, %s, %s, %s, %s);
+            ''', certificates)
+        dbconn.commit()
+    channel.basic_consume(
+        queue=queue_name,
+        on_message_callback=callback,
+        auto_ack=True)
+    channel.start_consuming()
+
+
+def main() -> None:
+    '''Main entrypoint.'''
+    parser = argparse.ArgumentParser(description=__doc__)
+    lib.db.configure_parser(parser)
+    lib.logs.configure_parser(parser)
+    parser.add_argument('--rabbitmq-user')
```

Comment:
```suggestion
    parser.add_argument('--rabbitmq-username')
```

aquí sólo se le está pasando el nombre de usuario, no el usuario.

Reply:
Listo